File: spring-data-multi-storage-transaction-sample/src/main/java/sample/command/GetCustomerInfoCommand.java
package sample.command;

import java.util.concurrent.Callable;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.SampleService;

@Component
@Command(name = "GetCustomerInfo", description = "Get customer information")
public class GetCustomerInfoCommand implements Callable<Integer> {
  @Autowired private SampleService sampleService;

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Override
  public Integer call() {
    System.out.println(sampleService.getCustomerInfo(customerId));
    return 0;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-sample/Commands/GetCustomerInfoCommand.cs
using System.CommandLine;

namespace ScalarDbClusterSample.Commands;

public static class GetCustomerInfoCommand
{
    private const string Name = "GetCustomerInfo";
    private const string Description = "Get customer information";

    private const string ArgName = "id";
    private const string ArgDescription = "customer ID";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName, ArgDescription);
        var getCustomerInfoCommand = new Command(Name, Description)
        {
            customerIdArg
        };

        getCustomerInfoCommand.SetHandler(async customerId =>
        {
            using var sample = new Sample();
            var customerInfo = await sample.GetCustomerInfo(customerId);
            
            Console.WriteLine(customerInfo);
        }, customerIdArg);
        
        return getCustomerInfoCommand;
    }
}
